### DISCUSSION TITLE
Token support, and support for adding multiple tokens

### DISCUSSION THREAD

**Discussion by Root-FTW** [AstraSurge/gpteams#55] [Ideas]
In the initial Chanzhaoyu project, it supports tokens, it would be great if this new project would also continue to support tokens..

other than that it would be great if you could add multiple tokens, for example I have 5 chatgpt plus accounts, and I would like to add the tokens of all 5 accounts.

Here is an example:
![2023_04_05_17_45_07](https://user-images.githubusercontent.com/108767514/230245847-962d604b-d2bb-434d-9c1c-a83a17857ade.png)

As you can see you could add a section for tokens, and this section we could add 1 or multiple tokens.

As you know when using tokens there is a limitation, if someone is currently using the chat no other user can use it since it can only be used once at a time, this is a natural limitation of the token.

In the other project a user already solved this limitation:
https://github.com/Chanzhaoyu/chatgpt-web/pull/925

My perfect idea would be that if I put 5 tokens, and a user when trying to use the chat the code detects that you can not use that token because someone else is using it automatically try the next token, and if it is busy also try the next one and so continue until a token of the 5 are available, that would be much better idea than just show an error and stop the process.

Another detail is that the names for the models are different when using the token, for example:

![2023_04_05_17_49_42](https://user-images.githubusercontent.com/108767514/230246420-6489f3eb-0f95-48e2-8f9d-10d02659e115.png)

when using the token, the best gpt3.5 model (ChatGPT PLUS) is called "text-davinci-002-render-sha", you can see this in the GET request.

I'm sure this idea is great for this project, and I'm sure it's not difficult for you to implement this improvement.

Thanks for giving us this great project

**Comment by suikodev** [comment]
I think using an access token for third-party applications goes against OpenAI's policy, and ChatGPT shares my view.
<img width="1114" alt="image" src="https://user-images.githubusercontent.com/37903575/230536240-284eb0b8-53c2-4522-b3ae-3242873fa57f.png">
